build/tools/SharedCodeGenerator/Program.cs

```diff
@@ -106,30 +106,100 @@ private static void GenerateSupportedPlatformsReadmeFile(string repoDir)
                 foreach (var subDirPath in Directory.GetDirectories(platformsDir))
                 {
                     var subDirInfo = new DirectoryInfo(subDirPath);
+                    var platformSubDirPaths = Directory.GetDirectories(subDirPath);
                     var platformName = subDirInfo.Name;
-                    sw.WriteLine($"## {platformName}");
-                    sw.WriteLine();
-                    var versionFile = Path.Join(subDirPath, "versionsToBuild.txt");
-                    using (var reader = new StreamReader(versionFile))
+                    foreach (var platformSubDirPath in platformSubDirPaths)
+                    {
+                        var platformSubDirPathInfo = new DirectoryInfo(platformSubDirPath);
+
+                        // maven and composer exist within the java and php directories with
+                        // their own versions, and we want to include those in the file
+                        switch (platformSubDirPathInfo.Name)
+                        {
+                            case "versions":
+                                sw.WriteLine($"## {platformName}");
+                                sw.WriteLine();
+                                AddVersions(sw, platformSubDirPath, platformName);
+                                break;
+                            case "maven":
+                                sw.WriteLine($"## {platformName} maven");
+                                sw.WriteLine();
+                                AddVersions(sw, Path.Join(platformSubDirPath, "versions"), "maven");
+                                break;
+                            case "composer":
+                                sw.WriteLine($"## {platformName} composer");
+                                sw.WriteLine();
+                                AddVersions(sw, Path.Join(platformSubDirPath, "versions"), "composer");
```

Comment:
We only have those 2 since they are the only languages with their own version files embedded in a language directory. So the structure for php and java looks like

- php/
  - versions/
  - composer/
    - versions/
- java/
  - versions/
  - maven/
    - versions/